### PR TITLE
Hotfix winning conditions

### PR DESCRIPTION
**Pull request by MathieuTuli**
The `win_condition` property as part of the `generator.game.Game` class (found in `textworld/generator/game.py`) was calling the wrong member variable for the quests. This PR fixes this by calling `[Quest].win_events`, not `[Quest].winning_conditions` as it previously was. 

Further, mirrored the `win_condition` property to create the analogous `fail_condition` property, if that info is ever needed by the user.

Lastly, renamed `win_condition` -> `win_conditions` and `fail_condition` -> `fail_conditions` to more accurately represent their return values, which are a list of conditions (which are in fact, `generator.game.Events`)

Note (Sanity Check): `winning_conditions` isn't a member variable that exists, and only appears in this one file in the whole code base.
```Console
⏵ grep -r winning_conditions --include=\*.py
textworld/generator/game.py:        return [q.winning_conditions for q in self.quests]
```

### CODE REVIEW

**Comment by MarcCote** [comment]
Thanks for the PR. Can I ask what is your use case for those functions? Since there are not used anywhere else in the code, I'm very tempted to simply delete them, actually.

If you do have a use case, proper tests should be added.

File: textworld/generator/game.py [hunk 1]
```diff
@@ -560,9 +560,14 @@ def verbs(self) -> List[str]:
         return sorted(set(cmd.split()[0] for cmd in self.command_templates))
 
     @property
-    def win_condition(self) -> List[Collection[Proposition]]:
+    def win_conditions(self) -> List[Collection[Proposition]]:
```
Comment (MarcCote) [inline textworld/generator/game.py:563]: The typing needs to be changed too. I.e., `List[Tuple[Event]]`.

File: textworld/generator/game.py [hunk 2]
```diff
@@ -560,9 +560,14 @@ def verbs(self) -> List[str]:
...
         """ All win conditions, one for each quest. """
-        return [q.winning_conditions for q in self.quests]
+        return [q.win_events for q in self.quests]
+
+    @property
+    def fail_conditions(self) -> List[Collection[Proposition]]:
```
Comment (MarcCote) [inline textworld/generator/game.py:568]: The typing needs to be changed too. I.e., `List[Tuple[Event]]`.

**Review by MarcCote — CHANGES_REQUESTED** [review]

**Comment by MathieuTuli** [comment]
> Thanks for the PR. Can I ask what is your use case for those functions? Since there are not used anywhere else in the code, I'm very tempted to simply delete them, actually.
> 
> If you do have a use case, proper tests should be added.

My use case lies in some work I'm doing relating to goal-conditioned RL. Having ground truth propositional logic of goal states is a required feature. As far as I could investigate, this was is the only way to retrieve that information (please correct me if I'm wrong). Although the propositional logic return by these functions uses the game's object IDs to instead of their names, so a second ID-> name lookup is actually needed, but that's just an aside. With this second lookup, I can compare against the game's` EnvInfo.facts` set of propositions.

Regarding tests, were there any specific guidelines that needed to be followed? My use case is just as information retrieval.

**Comment by MarcCote** [comment]
@MathieuTuli here's what I suggest: https://github.com/MathieuTuli/TextWorld/pull/1

**Comment by MathieuTuli** [comment]
> @MathieuTuli here's what I suggest: [MathieuTuli#1](https://github.com/MathieuTuli/TextWorld/pull/1)

Agreed, I had this in mind but opted for the quick workaround, which in hindsight was the wrong decision. Merged your PR, changes should be reflected.

**Comment by MarcCote** [comment]
Alright. Waiting for all the tests to pass, then I will make a minor release.

Thank you for suggesting this new feature.

**Comment by MarcCote** [comment]
I just noticed the PEP8 check has failed. Here's the fix: https://github.com/MathieuTuli/TextWorld/pull/2

**Comment by MathieuTuli** [comment]
> I just noticed the PEP8 check has failed. Here's the fix: [MathieuTuli#2](https://github.com/MathieuTuli/TextWorld/pull/2)

I had noticed it too, made the change directly just before you mentioned this.

**Comment by MarcCote** [comment]
Just a quick update. It seems there's an issue fetching packages from inform7.com website for running the tests on Azure pipelines. I'll check tomorrow to see if that's still an issue.

**Review by MarcCote — APPROVED** [review]

**Comment by MarcCote** [comment]
@MathieuTuli FYI TextWorld 1.4.4 is now released on PyPi.

**Pull request merged**